cmd/download_headers.cpp

```diff
@@ -50,7 +50,10 @@ int main(int argc, char* argv[]) {
 
     CLI11_PARSE(app, argc, argv);
 
-    log::set_verbosity(log::Level::kTrace);
+    log::Settings settings;
+    settings.log_threads = true;
+    log::init(settings);
+    log::set_verbosity(log::Level::kInfo);
     log::tee_file(std::filesystem::path("downloader.log"));
```

Comment:
I'd suggest to to set the file name into settings directly and let `log::init(settings)` do the rest.
```
log::Settings settings;
settings.log_file = "downloader.log";
log::init(settings);
```

Reply:
BTW ... you don't set a CLI argument to set verbosity level : as a result, being defaulted to Info, all traces are not printed.

Reply:
I just introduced CLI argument for log verbosity level, as you suggested